tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -33,6 +35,102 @@ def generate_machine_decryption_key():
         # randomly generated decryption key for Functions key
         return str(hexlify(os.urandom(32)).decode()).upper()
 
+    class FunctionAppInfrastructureParameters:
+        def __init__(self, group_name, location, app_insights_location, storage_name,
+                     service_plan_name, sku_name, sku_tier, webapp_name):
+            self.group_name = group_name
+            self.location = location
+            self.app_insights_location = app_insights_location
+            self.storage_name = storage_name
+            self.service_plan_name = service_plan_name
+            self.sku_name = sku_name
+            self.sku_tier = sku_tier
+            self.webapp_name = webapp_name
+
+    def deploy_infrastructure(self, parameters):
+        # Check if RG exists
+        session = self.local_session
+        rg_client = session.client('azure.mgmt.resource.ResourceManagementClient')
+        if not rg_client.resource_groups.check_existence(parameters.group_name):
+            self.log.info("Deploying resource group %s" % (parameters.group_name))
+            rg_client.resource_groups.create_or_update(parameters.group_name,
+                                                       {'location': parameters.location})
+        else:
+            self.log.info("Found resource group %s" % (parameters.group_name))
+
+        # Storage account create function is async, wait for completion after
+        # other resources provisioned.
+        sm_client = session.client('azure.mgmt.storage.StorageManagementClient')
+        accounts = sm_client.storage_accounts.list_by_resource_group(parameters.group_name)
```

Comment:
We know from above if the resource group didn't exist then the account isn't either.